### PR TITLE
docs: add crane 101 example

### PR DESCRIPTION
**Pull request by djzager**

### CODE REVIEW

File: content/content/Examples/stateless-app-mirror.md [hunk 1]
```diff
@@ -0,0 +1,290 @@
...
+bash <(curl -s "https://raw.githubusercontent.com/konveyor/crane/main/hack/minikube-clusters-start.sh")
+```
+
+or kind with:
+
+```bash
+bash <(curl -s "https://raw.githubusercontent.com/konveyor/crane-runner/main/hack/kind-up.sh")
+```
+
+You will also need to install `crane`. See the [Installation Guide]({{<ref "/content/Getting Started/installation" >}}).
```
Comment (alaypatel07) [inline content/content/Examples/stateless-app-mirror.md:35]: image
the link to Installation guide is not working

Comment (djzager) [reply content/content/Examples/stateless-app-mirror.md:35]: Are you browsing the preview https://deploy-preview-9--crane-documentation.netlify.app/content/examples/stateless-app-mirror/ ?

Looks okay to me

**Review by alaypatel07 — APPROVED** [review]
looks great, added a small nit to fix

ACK

File: content/content/Examples/stateless-app-mirror.md (hunk 1 above)
Comment (alaypatel07) [reply content/content/Examples/stateless-app-mirror.md:35]: no, I was browsing the markdown file directly on GitHub. Works fine on the preview link

File: content/content/Examples/stateless-app-mirror.md [hunk 2]
```diff
@@ -0,0 +1,290 @@
...
+        ├── Service_guestbook_frontend.yaml
+        ├── Service_guestbook_redis-master.yaml
+        └── Service_guestbook_redis-slave.yaml
+
+4 directories, 24 files
+```
+
+**NOTE**
+
+What's happening here? `crane export` is using a discovery client to see all of
```
Comment (eriknelson) [inline content/content/Examples/stateless-app-mirror.md:125]: Is it accurate to say it's using a "discovery client"? (I'm not actually sure if a client exists internally), or is it more accurate to call it the server's discovery API?

File: content/content/Examples/stateless-app-mirror.md [hunk 3]
```diff
@@ -0,0 +1,290 @@
...
+be executing a handful of JSON patches that:
+
+1. Remove the UID
+1. Remove the resourceVersion
+1. Remove the creationTimestamp
+1. Remove the generation field
+1. Remove the status
+
+All of this data is leftovers from the "source" cluster, and we are removing
+them from the final manifests to make them applicable to the "destination"
```
Comment (eriknelson) [inline content/content/Examples/stateless-app-mirror.md:236]: Do we want to mention the manifests should now be cluster agnostic, and therefore also appropriately non-specific enough to be versioned and managed via gitops / CD?

**Review by eriknelson — COMMENTED** [review]
Do we want to include instructions in the pre-requisites for how to find `kustomize` and what it is? I think it's fair to assume the user have access to `kubectl`...but people may not already be familiar with kustomize? At least the folks we worked with on the last hackathon seemed to be similar.

Another thought too, this is getting submitted to the upstream docs, which I think makes sense, but do we now want to add an initial scenario link to the crane-runner repo that links to this as a first "chapter" in the "crane book of scenarios"?

File: content/content/Examples/stateless-app-mirror.md [hunk 4]
```diff
@@ -0,0 +1,290 @@
...
+previously exported.
+
+# Apply Manifests to "destination" Cluster
+
+Now that you have manifests prepared for the "destination" cluster, you can
+apply them using `kubectl` directly:
+
+```bash
+kubectl --context dest create namespace guestbook
+kubectl --context dest --namespace guestbook --recursive=true -f apply
```
Comment (eriknelson) [inline content/content/Examples/stateless-app-mirror.md]: I think this command assumes a working directory? Should we alter this to point -f to the output directory?

**Review by eriknelson — CHANGES_REQUESTED** [review]

File: content/content/Examples/stateless-app-mirror.md [hunk 5]
```diff
@@ -0,0 +1,290 @@
...
+If the flags start to get tough to manage via the command-line, you can also
+specify a `--flags-file` that might look something like:
+
+```yaml
+debug: false
+export-dir: myExport
+transform-dir: myTransform
+output-dir: myOutput
+optional-flags:
+  add-annotations: "custom-crane-annotation=foo"
```
Comment (sseago) [inline content/content/Examples/stateless-app-mirror.md]: Actually, this should be specified as a map:
```
  add-annotations:
    "custom-crane-annotation": "foo"
```

Comment (sseago) [reply content/content/Examples/stateless-app-mirror.md]: Map and slice params have to be encoded as comma-delimited strings when using a CLI flag, but in yaml they're specified as normal slices or maps.

**Review by sseago — CHANGES_REQUESTED** [review]
Looks good overall -- one minor correction in the flags-file example.

**Review by JaydipGabani — APPROVED** [review]

**Comment by JaydipGabani** [comment]
Didn't find anything bugging except the comments that are already made

File: content/content/Examples/stateless-app-mirror.md (hunk 2 above)
Comment (djzager) [reply content/content/Examples/stateless-app-mirror.md:125]: I called it a discovery client based on https://github.com/konveyor/crane/blob/27d247cbbe960c38d5be331e9983d0192601560b/cmd/export/export.go#L96-L100

```
	discoveryClient, err := o.configFlags.ToDiscoveryClient()
	if err != nil {
		log.Errorf("cannot create discovery client: %#v", err)
		return err
	}
```

Comment (eriknelson) [reply content/content/Examples/stateless-app-mirror.md:125]: +1 thought that might have been the case and I was unfamiliar. LGTM as is.

File: content/content/Examples/stateless-app-mirror.md [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+specify a `--flags-file` that might look something like:
+
+```yaml
+debug: false
+export-dir: myExport
+transform-dir: myTransform
+output-dir: myOutput
+optional-flags:
+  add-annotations:
+    custom-crane-annotation: "foo"
```
Comment (sseago) [inline content/content/Examples/stateless-app-mirror.md:203]: Not sure whether it's required, but I think the annotation name (custom-crane-annotation) should probably be in quotes here.

**Review by sseago — COMMENTED** [review]
One more nit, then I think we're good to go

File: content/content/Examples/stateless-app-mirror.md (hunk 6 above)
Comment (djzager) [reply content/content/Examples/stateless-app-mirror.md:203]: Using this config file worked for me, and it applied the correct annotations ie:

```
apiVersion: v1
kind: Service
metadata:
  annotations:
    custom-crane-annotation: foo
```

File: content/content/Examples/stateless-app-mirror.md (hunk 5 above)
Comment (sseago) [reply content/content/Examples/stateless-app-mirror.md]: Yeah, actually the quotes aren't required here. I was getting confused between yaml and json here. I think this is good.

**Review by sseago — APPROVED** [review]

**Pull request merged**